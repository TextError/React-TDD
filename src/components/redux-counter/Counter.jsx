import React from 'react';
import { connect } from 'react-redux';

const Counter = ({ counter }) => {

  const onIncrement = () => {};
  const onDecrement = () => {};

  return (
    <div>
      <h2>Counter</h2>
      <div>
        <button data-testid="button-decrement" onClick={onDecrement}>down</button>
        <span data-testid="counter">{counter}</span>
        <button data-testid="button-increment" onClick={onIncrement}>up</button>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  counter: state.counter.counter
})

export default connect(mapStateToProps, {})(Counter);