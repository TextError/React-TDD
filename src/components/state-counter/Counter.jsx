import React, { useState } from 'react';

const Counter = () => {
  const [state, setState] = useState({ count: 0 });
  const { count } = state;

  const onIncrement = () => setState({ count: count+1 });
  const onDecrement = () => setState({ count: count-1 });
  const onDelay = () => setTimeout(() => setState({ count: count+1 }), 1000);

  return (
    <div>
      <h2>Counter</h2>
      <div>
        <button data-testid="button-decrement" onClick={onDecrement}>down</button>
        <span data-testid="count">{count}</span>
        <button data-testid="button-increment" onClick={onIncrement}>up</button>
        <button data-testid="button-up-await" onClick={onDelay}>delay</button>
      </div>
    </div>
  );
};

export default Counter;