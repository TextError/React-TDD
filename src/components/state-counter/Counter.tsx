import React, { useState } from 'react';

const Counter = () => {
  const [state, setState] = useState({ count: 0 });
  const { count } = state;

  const onIncrement = () => setState({ count: count+1 });
  const onDecrement = () => setState({ count: count-1 });

  return (
    <div>
      <h2>Counter</h2>
      <div>
        <button data-testid="button-decrement" onClick={onDecrement}>-</button>
        <span data-testid="count">{count}</span>
        <button data-testid="button-increment" onClick={onIncrement}>+</button>
      </div>
    </div>
  );
};

export default Counter;