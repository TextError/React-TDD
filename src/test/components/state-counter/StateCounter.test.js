import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Counter from '../../../components/state-counter/Counter';

afterEach(cleanup);

describe('Check snapshot of the component', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Counter />)
    
    expect(asFragment(<Counter />)).toMatchSnapshot()
   })
});