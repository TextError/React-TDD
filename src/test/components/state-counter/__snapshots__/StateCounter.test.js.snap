// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check snapshot of the component should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Counter
    </h2>
    <div>
      <button
        data-testid="button-decrement"
      >
        -
      </button>
      <span
        data-testid="count"
      >
        0
      </span>
      <button
        data-testid="button-increment"
      >
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;
